### ISSUE TITLE
racket-trim-requires does not trim 'racket' or 'racket/base'

### ISSUE THREAD

**Issue by wolverian** [greghendershott/racket-mode#117]
When in a module:

``` racket
#lang racket

(require racket)
```

`M-x racket-trim-requires` does not trim `(require racket)`, even though it is redundant. The same applies when the module is `#lang racket/base` and requires `racket/base` separately. I guess that any module that is required twice – once in the `#lang` pragma and once in a `require` – is not trimmed.

**Comment by greghendershott** [comment]
The `racket-trim-requires` and `racket-base-requires` commands use the analysis done by [macro-debugger/analysis/check-requires](http://docs.racket-lang.org/macro-debugger/index.html#%28def._%28%28lib._macro-debugger%2Fanalysis%2Fcheck-requires..rkt%29._show-requires%29%29), which also has a command-line interface -- if you run that file through `raco check-requires`, it reports nothing should change.

In other words we should probably file this (and also issue #76) against the racket/macro-debugger repo [here](https://github.com/racket/macro-debugger/issues). If you have time to do so, that would be great (you could simply link to the issues here). Otherwise I'll do it.  Ideally I'd figure out the issue and submit a PR there, but I'm not sure if I have time now to grok it.

**Comment by greghendershott** [comment]
From the discussion there it looks like this would need to be fixed here in racket-mode. Although I don't feel like this is an awful/urgent thing fix (let me know if you disagree), I'll tackle it when my queue of higher-priority things gets smaller. Thank you for reporting it, as well as for opening the issue over there.

**Comment by wolverian** [comment]
Thanks! I agree that this is more of a quality of life issue than a blocker.

**Comment by greghendershott** [comment]
See also #76 and racket/macro-debugger#3.
